fgpyo/sequence.py

```diff
@@ -56,6 +58,19 @@
 }
 
 
+# Get str of all invalid nucleotide characters (strings missing from _COMPLEMENTS).
+_INVALID_BASES: str = "".join(c for c in (chr(o) for o in range(256)) if c not in _COMPLEMENTS)
+# Use str.maketrans to create a table matching ascii codes of bases to ascii codes of complements.
+_COMPLEMENTS_TABLE: MappingProxyType[int, Optional[int]] = MappingProxyType(
+    str.maketrans("".join(_COMPLEMENTS.keys()), "".join(_COMPLEMENTS.values()), _INVALID_BASES)
+)
+"""
+This table allows faster reverse-complement than directly using _COMPLEMENTS iterating over bases.
+Characters from _INVALID_BASES will be mapped to None and omitted from translation, which can be
+detected as an error by noting decreased string length.
+"""
+
+
 def complement(base: str) -> str:
```

Comment:
Odd that we have a `complement()` that only takes 1bp sequences.

And a `reverse_complement()` that takes n-length sequences.